src/cephadm/cephadm

```diff
@@ -2452,7 +2487,7 @@ def command_deploy():
             daemon_ports = Monitoring.port_map[daemon_type]  # type: List[int]
             if any([port_in_use(port) for port in daemon_ports]):
                 raise Error("TCP Port(s) '{}' required for {} is already in use".format(",".join(map(str, daemon_ports)), daemon_type))
-            elif args.image == DEFAULT_IMAGE or args.image == inferred_image:
+            elif args.image == DEFAULT_IMAGE:
```

Comment:
@liewegas Since `@require_image` will guarantee that an image was already supplied, I wonder if this error message can be improved. It will look like a bug if the user sees the `--image parameter must be supplied for {}` message.